lib/audio/client/audio-ui.js

```diff
@@ -26,27 +25,27 @@ class AudioUi
         if(!this.audioManager.categories){
             return;
         }
-        let audioSettingsTemplate = this.uiScene.cache.html.get('audio');
-        let audioCategoryTemplate = this.uiScene.cache.html.get('audio-category');
+        let audioSettingsTemplate = this.uiSceneManager.uiSceneDriver.getCacheHtml('audio');
+        let audioCategoryTemplate = this.uiSceneManager.uiSceneDriver.getCacheHtml('audio-category');
         let audioSettingsContent = this.prepareAudioSettingsContent(audioCategoryTemplate, audioSettingsTemplate);
-        this.gameManager.gameDom.appendToElement('#settings-dynamic', audioSettingsContent);
-        let audioSettingInputs = this.gameManager.gameDom.getElements('.audio-setting');
+        this.uiSceneManager.gameDom.appendToElement('#settings-dynamic', audioSettingsContent);
+        let audioSettingInputs = this.uiSceneManager.gameDom.getElements('.audio-setting');
         if(0 === audioSettingInputs.length){
             return false;
         }
         for(let settingInput of audioSettingInputs){
             settingInput.addEventListener('click', async (event) => {
                 await this.audioManager.setAudio(event.target.dataset['categoryKey'], settingInput.checked);
-                this.sendAudioUpdate(settingInput.value, settingInput.checked, this.gameManager.activeRoomEvents);
-                this.sceneAudioPlayer.playSceneAudio(this.audioManager, this.gameManager.getActiveScene());
+                this.sendAudioUpdate(settingInput.value, settingInput.checked, this.uiSceneManager.getActiveRoomEventsCallback());
+                this.sceneAudioPlayer.playSceneAudio(this.audioManager, this.uiSceneManager.getActiveSceneCallback());
             });
         }
     }
 
     prepareAudioSettingsContent(audioCategoryTemplate, audioSettingsTemplate)
     {
         let categoriesRows = this.prepareCategoriesRows(audioCategoryTemplate);
-        return this.gameManager.gameEngine.parseTemplate(audioSettingsTemplate, {audioCategories: categoriesRows});
+        return this.uiSceneManager.parseTemplateCallback(audioSettingsTemplate, {audioCategories: categoriesRows});
```

Comment:
the parseTemplateCallback is calling parseTemplate on the gameEngine, but instead of passing the whole object through there is better to send only a callback. At the end, both are calling the same method, but my implementation is passing only a callback and the previous lean on sending a whole object.